redfish-core/lib/assembly.hpp

```diff
@@ -0,0 +1,493 @@
+            // implemeted. This is to handle the case in which there is
+            // entry in association json but implementation of interface for
+            // that particular assembly is missing.
+            auto it = std::find(assemblies.begin(), assemblies.end(),
+                                objectPath);
+            if (it != assemblies.end())
+            {
+                updatedAssemblyList.emplace(updatedAssemblyList.end(), *it);
+            }
+        }
+
+        if (!updatedAssemblyList.empty())
+        {
+            // sorting is required to facilitate patch as the array does not
+            // store and data which can be mapped back to Dbus path of
+            // assembly.
+            std::sort(updatedAssemblyList.begin(), updatedAssemblyList.end());
+
+            getAssemblyProperties(asyncResp, chassisPath, updatedAssemblyList);
+        }
+    });
+}
+
+/**
+ * @brief Api to get assembly endpoints from mapper.
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisPath - Chassis to which the assemblies are
+ * associated.
+ *
+ * @return None.
+ */
+inline void
+    getAssemblyEndpoints(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const std::string& chassisPath)
+{
+    BMCWEB_LOG_DEBUG("Get assembly endpoints");
+
+    sdbusplus::message::object_path assemblyPath(chassisPath);
+    assemblyPath /= "assembly";
+
+    // if there is assembly association, look
+    // for endpoints
+    dbus::utility::getAssociationEndPoints(
+        assemblyPath, [asyncResp, chassisPath](
+                          const boost::system::error_code& ec,
+                          const dbus::utility::MapperEndPoints& endpoints) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error");
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        auto sortedAssemblyList = endpoints;
+        std::sort(sortedAssemblyList.begin(), sortedAssemblyList.end());
+
+        checkAssemblyInterface(asyncResp, chassisPath, sortedAssemblyList);
+        return;
+    });
+}
+
+/**
+ * @brief Api to check for assembly associations.
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisPath - Chassis to which the assemblies are
+ * associated.
+ *
+ * @return None.
+ */
+inline void checkForAssemblyAssociations(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisPath, const std::string& service)
+{
+    BMCWEB_LOG_DEBUG("Check for assembly association");
+
+    using associationList =
+        std::vector<std::tuple<std::string, std::string, std::string>>;
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp,
+         chassisPath](const boost::system::error_code ec,
+                      const std::variant<associationList>& associations) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error");
```

Comment:
Should these BMCWEB_LOG_DEBUG() for internalError() be BMCWEB_LOG_ERROR(), and include `ec.value()?

```
            BMCWEB_LOG_ERROR("DBUS response error {}", ec.value());
            messages::internalError(asyncResp->res);
```